src/js/modules/ps_mainmenu.ts

```diff
@@ -1,45 +1,91 @@
- *
- * @author    PrestaShop SA and Contributors <contact@prestashop.com>
- * @copyright Since 2007 PrestaShop SA and Contributors
- * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
  */
+
 import {Dropdown} from 'bootstrap';
 
-const initDropdownToggleMenu = () => {
+const ESCAPE_KEY = 'Escape';
+const ARROW_UP_KEY = 'ArrowUp';
+const ARROW_DOWN_KEY = 'ArrowDown';
+const ARROW_LEFT_KEY = 'ArrowLeft';
+const ARROW_RIGHT_KEY = 'ArrowRight';
+
+const initDesktopMenu = () => {
   const {prestashop} = window;
-  const {dropdownToggles: dropdownTogglesMap} = prestashop.themeSelectors;
-  const dropdownToggles = document.querySelectorAll(dropdownTogglesMap.dropdownToggles);
+  const {desktopMenu: desktopMenuMap} = prestashop.themeSelectors;
 
-  dropdownToggles.forEach((el) => {
-    const dropdown = new Dropdown(el, {
+  /**
+   * Handle Mouse and Keyboard events for Submenus.
+   * Find all dropdown toggles with [data-depth: ^2]
+   */
+  const dropdownToggles = document.querySelectorAll(desktopMenuMap.dropdownToggles);
+  dropdownToggles.forEach((dropdownToggleElement: HTMLElement) => {
+    const dropdown = new Dropdown(dropdownToggleElement, {
       popperConfig: () => ({placement: 'auto-end'}),
     });
-    el.parentElement?.addEventListener('mouseover', () => {
+
+    /* Control Submenu display for Mouse events inside the menu container */
+    const dropdownElement = dropdownToggleElement.parentElement;
+    dropdownElement?.addEventListener('mouseover', () => {
       dropdown.show();
+      // Revert auto-focus by BS on dropdown toggle to controll Keyboard events for Submenu
+      dropdownToggleElement.blur();
     });
-    el.parentElement?.addEventListener('mouseout', () => {
+    dropdownElement?.addEventListener('mouseout', () => {
       dropdown.hide();
     });
+
+    /* Control Submenu display for Keyboard events inside the menu container */
+    dropdownToggleElement.addEventListener('keydown', (event: KeyboardEvent) => {
+      // Open the submenu when Arrow Right or Arrow Left key is selected
+      if (event.key === ARROW_LEFT_KEY || event.key === ARROW_RIGHT_KEY) {
+        dropdown.show();
+      }
+
+      // Hide the submenu when Escape key is pressed
+      if (event.key === ESCAPE_KEY) {
+        dropdown.hide();
+      }
+    });
+  });
+
+  /**
+   * Handle Keyboard events for main menu links
+   * Find all links with [data-depth: 0]
+   */
+  const mainMenuLinks = document.querySelectorAll(desktopMenuMap.dropdownItemAnchor(0));
+  mainMenuLinks.forEach((mainMenuLinkElement: HTMLElement) => {
+    const menuContainerElement = mainMenuLinkElement.nextElementSibling;
+
+    /* Open the menu container when Arrow Up or Arrow Down key is pressed */
+    mainMenuLinkElement.addEventListener('keydown', (event: KeyboardEvent) => {
```

Comment:
Do we really need this? Bootstrap dropdowns should naturally handle keyboard navigation

Reply:
The current structure is not based on BS standard for dropdowns,
If we add 'dropdown' selector to main links parent container to enable BS keyboard navigation,
then we will have problem for controlling the menu container, so need to do a refacto and change current structure to BS navbar.

Reply:
Do you see some constraints on moving to the BS structure? If we can use the exact same style but with a cleaner bootstrap structure, that could be a great maintainable solution

I would probably accept your PR as a first workaround, but if we can do something more cleaner, sticking to BS, maybe this is the right choice

wdyt @Hlavtox ?

Reply:
The current structure is fine,
If you try to use BS Navbar with dropdowns in main links, then you can not have menu-container with rows and columns.

In this PR I tried to simulate BS dropdown behavior for dropdowns in main-links.